### PR TITLE
fix!: fix course detail page url bug

### PR DESCRIPTION
**Pull request by Faraz32123**
Opened this PR with a fix that @regisb suggested for this [issue](https://github.com/openedx/wg-build-test-release/issues/301).

### CODE REVIEW

File: ecommerce/static/js/models/course_model.js [hunk 1]
```diff
@@ -31,6 +31,9 @@ define([
 
         return Backbone.RelationalModel.extend({
             urlRoot: '/api/v2/courses/',
+            url: function() {
```
Comment (regisb) [inline ecommerce/static/js/models/course_model.js:36]: Maybe add a comment to explain why we need to do this?

**Comment by antoviaque** [comment]
@mphilbrick211 @regisb Following up on the discussion from https://discuss.openedx.org/t/contributors-meetup-async-update-september-2nd-2023-september-15th-2023/11185/19 as suggested

> I don’t think we’d want to convert it if it’s not truly open-source, as there’s lots of PRs worked on by 2U and the community. It might be better to communicate directly with the person who needs to review it.

@regisb You mentioned that this is should actually be an OSPR - is it because it originates from you, or that @Faraz32123 you meant to open this as an OSPR?

**Comment by regisb** [comment]
I'm not sure what is meant by "truly open source". If the question is whether this PR was opened on behalf of 2U, then no, that's not the case. Faraz is making this contribution on behalf of Edly, to resolve a problem for the community. For all means and purposes, this PR can be considered as a free, open source contribution.

Is there a bigger context that I am missing here? Would there be the same question if I had opened the PR myself?

**Comment by antoviaque** [comment]
@regisb 

> Is there a bigger context that I am missing here? Would there be the same question if I had opened the PR myself?

As far as I understand that's what created the confusion yes - but @mphilbrick211 might be able to confirm.

**Comment by mphilbrick211** [comment]
@antoviaque @Faraz32123 - I checked in on this with Ned B. to see if there was an issue with the bot. We don't see any issues, so suspect maybe there was a brief outage and this pull request never got labeled. I've added it to the contributions board now.

**Comment by mphilbrick211** [comment]
Hi @Faraz32123! Looks like there's a failing check. Would you mind taking a look?

**Comment by regisb** [comment]
I'm pretty sure that the build issue has nothing to do with the changes in this PR: https://readthedocs.org/projects/edx-ecommerce/builds/22486927/ @Faraz32123 you should just have to rebase your changes on top of master to fix the build.

**Comment by regisb** [comment]
@Faraz32123 what's the status of this PR? Can you please rebase on top of master?

File: ecommerce/static/js/models/course_model.js [hunk 2]
```diff
@@ -31,6 +31,11 @@ define([
 
         return Backbone.RelationalModel.extend({
             urlRoot: '/api/v2/courses/',
+            // course detail page url was not matching because of missing trailing slash,
+            // with thi function, trailing slash will be added to match the url
```
Comment (feanil) [inline ecommerce/static/js/models/course_model.js:35]:
```suggestion
            // with this function, trailing slash will be added to match the url
```

**Comment by feanil** [comment]
@Faraz32123 a small grammar change that I can't make to your PR but if you accept that, I can merge this.

**Comment by mariajgrimaldi** [comment]
@feanil: is there something we can do to merge this or should we close the PR considering it'd archived?

**Pull request closed without merging**